### PR TITLE
Update pipeline template annotations

### PR DESCRIPTION
**Pull request by spadgett**
* Move Jenkins pipeline template service annotations to correct resource
* Add pipeline.alpha.openshift.io/uses to the pipeline BuildConfig

@bparees PTAL
@jwforres CC

### CODE REVIEW

**Comment by bparees** [comment]
needs hack/gen-bootstrap run

**Comment by spadgett** [comment]
thanks @bparees another look?

**Comment by bparees** [comment]
lgtm [merge]

**Comment by spadgett** [comment]
@bparees Spoke to @smarterclayton. We want to use `pipeline.alpha.openshift.io/uses` for the build config annotation. Added it to the PR.

@smarterclayton Look right?

File: examples/jenkins/pipeline/pipelinetemplate.json [hunk 1]
```diff
@@ -19,6 +19,9 @@
         "creationTimestamp": null,
         "labels": {
           "name": "sample-pipeline"
+        },
+        "annotations": {
+          "pipeline.alpha.openshift.io/uses": "[{\"name\": \"frontend\", \"namespace\": \"\", \"kind\": \"DeploymentConfig\"}]"
```
Comment (bparees) [inline examples/jenkins/pipeline/pipelinetemplate.json]: is this supplanting this pr:
https://github.com/openshift/origin/pull/9275
?

Comment (spadgett) [inline examples/jenkins/pipeline/pipelinetemplate.json]: Yes.

**Pull request merged**